assets/components/ovn/master/daemonset.yaml

```diff
@@ -120,32 +120,26 @@ spec:
           trap quit TERM INT
 
           bracketify() { case "$1" in *:*) echo "[$1]" ;; *) echo "$1" ;; esac }
+          compact() {
+            while true; do
+              /usr/bin/ovn-appctl -t /var/run/ovn/ovn${1}_db.ctl --timeout=5 ovsdb-server/compact 2>/dev/null || true
+              sleep 600
+            done
+          }
           # initialize variables
           db="nb"
           ovn_db_file="/etc/ovn/ovn${db}_db.db"
-
           OVN_ARGS="--db-nb-cluster-local-port=9643 --no-monitor"
 
           echo "$(date -Iseconds) - starting nbdb"
-          initialize="false"
-
-          if [[ ! -e ${ovn_db_file} ]]; then
-            initialize="true"
-          fi
+          exec /usr/share/ovn/scripts/ovn-ctl \
+            ${OVN_ARGS} \
+            --ovn-nb-log="-vconsole:${OVN_LOG_LEVEL} -vfile:off -vPATTERN:console:%D{%Y-%m-%dT%H:%M:%S.###Z}|%05N|%c%T|%p|%m" \
+            run_nb_ovsdb &
 
-          if [[ "${initialize}" == "true" ]]; then
-                exec /usr/share/ovn/scripts/ovn-ctl ${OVN_ARGS} \
-                --ovn-nb-log="-vconsole:${OVN_LOG_LEVEL} -vfile:off -vPATTERN:console:%D{%Y-%m-%dT%H:%M:%S.###Z}|%05N|%c%T|%p|%m" \
-                run_nb_ovsdb &
-
-                wait $!
-          else
-            exec /usr/share/ovn/scripts/ovn-ctl ${OVN_ARGS} \
-              --ovn-nb-log="-vconsole:${OVN_LOG_LEVEL} -vfile:off -vPATTERN:console:%D{%Y-%m-%dT%H:%M:%S.###Z}|%05N|%c%T|%p|%m" \
-              run_nb_ovsdb &
-
-              wait $!
-          fi
+          db_pid=$!
+          compact $db &
```

Comment:
I guess there is no need to wait for the compact process, wdyt?

Reply:
It's not waiting, just running it in the background. So to answer Miguel's question:
> I wonder if the while loop will exit on the signal to stop the process, probably yes.

I'd say that it'll exit just fine because we're not waiting on it, just running it in the background and when db exits (wait $db), this background subprocess will go down as well

Reply:
makes sense